movement/analysis/kinematics.py

```diff
@@ -165,3 +169,472 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
+        :func:`scipy.spatial.distance.cdist`.
+
+    Returns
+    -------
+    xarray.DataArray
+        An xarray DataArray containing the computed distances between
+        each pair of inputs.
+
+    Examples
+    --------
+    Compute the Euclidean distance (default) between ``ind1`` and
+    ``ind2`` (i.e. interindividual distance for all keypoints)
+    using the ``position`` data variable in the Dataset ``ds``:
+
+    >>> pos1 = ds.position.sel(individuals="ind1")
+    >>> pos2 = ds.position.sel(individuals="ind2")
+    >>> ind_dists = cdist(pos1, pos2, dim="individuals")
+
+    Compute the Euclidean distance (default) between ``key1`` and
+    ``key2`` (i.e. interkeypoint distance for all individuals)
+    using the ``position`` data variable in the Dataset ``ds``:
+
+    >>> pos1 = ds.position.sel(keypoints="key1")
+    >>> pos2 = ds.position.sel(keypoints="key2")
+    >>> key_dists = cdist(pos1, pos2, dim="keypoints")
+
+    Obtain the distance between ``key1`` of ``ind1`` and
+    ``key2`` of ``ind2`` from ``ind_dists``
+    (i.e. interindividual distance, different keypoints):
+
+    >>> dist_ind1key1_ind2key2 = ind_dists.sel(ind1="key1", ind2="key2")
+
+    Equivalently, the same distance can be obtained from ``key_dists``:
+
+    >>> dist_ind1key1_ind2key2 = key_dists.sel(key1="ind1", key2="ind2")
+
+    Obtain the distance between ``key1`` and ``key2`` of ``ind1``
+    (i.e. interkeypoint distance within the same individual):
+
+    >>> dist_ind1key1_ind1key2 = key_dists.sel(key1="ind1", key2="ind1")
+
+    Obtain the distance between ``key1`` of ``ind1`` and ``ind2``
+    (i.e. interindividual distance, same keypoint):
+
+    >>> dist_ind1key1_ind2key1 = ind_dists.sel(ind1="key1", ind2="key1")
+
+    See Also
+    --------
+    scipy.spatial.distance.cdist : The underlying function used.
+    compute_interindividual_distances : Compute distances between one or
+        more pairs of individuals within and across all keypoints.
+    compute_interkeypoint_distances : Compute distances between one or
+        more pairs of keypoints within and across all individuals.
+
+    """
+    core_dim = "individuals" if dim == "keypoints" else "keypoints"
+    elem1 = getattr(a, dim).item()
+    elem2 = getattr(b, dim).item()
+    a = _validate_core_dimension(a, core_dim)
+    b = _validate_core_dimension(b, core_dim)
+    result = xr.apply_ufunc(
+        _cdist,
+        a,
+        b,
+        kwargs={"metric": metric, **kwargs},
+        input_core_dims=[[core_dim, "space"], [core_dim, "space"]],
+        output_core_dims=[[elem1, elem2]],
+        vectorize=True,
+    )
+    result = result.assign_coords(
+        {
+            elem1: getattr(a, core_dim).values,
+            elem2: getattr(a, core_dim).values,
+        }
+    )
+    # Drop any squeezed coordinates
+    return result.squeeze(drop=True)
+
+
+def compute_interindividual_distances(
+    data: xr.DataArray,
+    pairs: dict[str, str | list[str]] | None = None,
```

Comment:
I worry about `None` being the default - meaning that it will by default calculate all possible pairs. Combinatorial explosion means the numbers can get large very fast. For example, when I tried it on the crab sample dataset (which has many individuals), the computation was noticably slow and I got nearly 4k pairwise distances as output! For a long video with high fps this could have led to a crash.The same problem arises for `compute_interkeypoint_distances` if a dataset has many keypoints.

A potential way forwards (open to counter-proposals):
- Make `pairs` a required argument, so the user has to explicitly request only what they need
- For cases where users actually need all possible pairs, allow for a special `pairs="all"` argument which will produce the behaviour we currently get with `pairs=None`. So if they want everything, they have to explicitly say so.